### ISSUE TITLE
Enable partial formatting (maybe using formatting tags)

### ISSUE THREAD

**Issue by berry120** [quelea-projection/Quelea#29]
At present all text in a song is presented in the same font / style (with the exception of chords and translations.)

We've had requests for partial formatting a few times now - the simplest way of doing this may well be using formatting tags in a similar vein to OpenLP (https://manual.openlp.org/display_tags.html) but there may be other approaches as well.

**Comment by berry120** [comment]
*Still considering whether this is something worth adding.* This may well be a pain to implement in terms of alignment, lyrics, etc. and hasn't been requested too much in the past. Give this issue a thumbs up if you'd like to see it implemented.

**Comment by hosios** [comment]
I would like to present the chorus or bridge in italics for example.

**Comment by archenians** [comment]
Thank you for considering my request, i'm still waiting the app will support the underline formatting.

**Comment by ipikles** [comment]
I don't see much worth in that feature either. However, if you choose to implement here are some thoughts:

### Colors
In OpenLP you can format text to a certain color, but color is always "hard coded". For example, you can set texts color to red or blue. But what happens if you force text to blue and used theme just happens to have blue background?

For those cases I propose that colors are given class names. Those colors are then defined in theme so themes creator can make sure that all colors are easy to read. Example class names could be "highlight", "primary" and "secondary".

For example
<pre>
Line of text
{secondary}Other line of text{/secondary}
</pre>

Actual use case might be songs where different groups of people will sing different parts of the song. [Me syömme leipää elämän](https://fi.scoutwiki.org/Ruokalaulut#Me_sy.C3.B6mme_leip.C3.A4.C3.A4_el.C3.A4m.C3.A4n) is a classic example where girls ask questions and boys answers (or vice versa). (This can of course be done with italics or similar methods)

### Position of the chords
I don't know is this an issue, but lets assume you have word that has over 40 letters and you make it bold. Could that word grow that much that chords are no longer in correct places? I know English is language of short words, but you will see longer words in other languages :)

### Markdown syntax or similar?
If we don't need colors then `_underline_`, `*italic*` and `**bold**` might be enought. They are easy and quick enough to write when translating songs and doesn't consume much space.

**Comment by berry120** [comment]
Thanks for the thoughts.

We did consider markdown a while back, but came to the conclusion that someone would always want something more than markdown could provide, and then we'd be back to square one again (or close to it.)

**Comment by hosios** [comment]
I agree with @ArvidNy: less distraction in worship is better. I'm coming from 10 years of using OpenSong, and they have an option on the song theme to click if you want the chorus in italics. And at our church we are used to it now. But it is not a killer feature. Probably better to invest the coding time in something else.

**Comment by archenians** [comment]
So, the conclusion is underline partial formatting won't be provide in the future?
Unfortunately, even though many churches worshipper in my country who need it.
But as a developer, you could reject it of course

**Comment by berry120** [comment]
It's unlikely to be a priority, but that doesn't mean it won't be provided
in the future. We'll re-evaluate after 2019.0 is out most likely and take
it from there.

**Comment by berry120** [comment]
This remains on the cards, but no work has been completed on it yet.

The real challenge here isn't technical, it's how it should work in the interface - I've not yet been convinced that any given approach is ideal. Formatting tags are a possibility, but it adds complexity when entering a song, and colours in particular are a challenge as they'd have to change per theme - we'd need a "primary" and "secondary" colour for text for instance, which also adds more complexity both from a user experience and coding point of view.

Still open to hearing further ideas on this one as always, but at the moment it's looking like it won't be implemented any time soon - sorry about that.

**Comment by c-exe** [comment]
Implementing this would remove the need for the [Hack](https://quelea-projection.github.io/docs/Minister_Response_Hack) for call-and-response, as then it would be simple to copy in the liturgy as-is and then apply the second formatting style to just the required lines.  Using the hack is a lot of work if there is a lot of liturgy to put in (such as for communion or a baptism).
